src/resspect/fit_lightcurves.py

```diff
@@ -323,29 +323,32 @@ def _sample_fit(
     light_curve_data.id = obj_dic['objectid']
     light_curve_data.redshift = obj_dic['redshift']
     light_curve_data.sncode = obj_dic['sncode']
-    if light_curve_data.sncode in Ia_code:
-        light_curve_data.sntype = 'Ia'
+    if light_curve_data.sncode in one_code:
+        light_curve_data.sntype = 'Ia' #just labeling all positive classes as Ia
+                                       #unsure what changing this might affect in database.py
     else:
         light_curve_data.sntype = 'other'
     light_curve_data.sample = type
-    light_curve_data.additional_info = additional_info
-
+    light_curve_data.additional_info = []
+    for info in additional_info:
+        light_curve_data.additional_info.append(obj_dic[info])
     light_curve_data.fit_all()
     
     return light_curve_data
 
-def fit(data_dic: dict, output_features_file: str,
+def fit(data_dic: list, output_features_file: str,
             number_of_processors: int = 1,
             feature_extractor: str = 'bazin', filters: list = ['SNPCC'], 
-            features: list = [], type: str = 'unspecified', Ia_code: list = [10],
+            features: list = [], type: str = 'unspecified', one_code: list = [10],
             additional_info: list = []):
     """
     Perform fit to all objects from a generalized dataset.
 
      Parameters
      ----------
-     data_dic: str
-         Dictionary containing the photometry for all light curves.
+     data_dic: list
```

Comment:
```suggestion
     data_dic: list[dict]
```

Minor point - doesn't need to be fixed in this PR - but since `data_dic` is now expected to be a `list`, it might make sense to simply call it `data`.